### ISSUE TITLE
SA1629 for multiline exception documentation

### ISSUE THREAD

**Issue by paulomorgado** [DotNetAnalyzers/StyleCopAnalyzers#2712]
SA1629 is being flagged at `-or-`.

```C#
/// <exception cref="System.ArgumentException">
/// <para>If settings are required and <paramref name="settings" /> is <see langword="null" />.</para>
/// -or-
/// <para>If any of the required settings is not in <paramref name="settings" /> or has invalid values.</para>
/// </exception>
```

I'm using the usual MSDN documentation style. Is there another recommendation? Or is this a bug?

**Comment by sharwell** [comment]
I would suggest you gather a list of cases where SA1629 was reported unexpectedly, and then file bugs for the high-level issues. Then disable the rule for now. Hopefully it will make more sense to enable it after we're able to address the bugs from the first round.

**Comment by paulomorgado** [comment]
So far, this is the only one I've came across.

I'm trying to follow the style of MSDN documentations, although multiple entries are allowed for the same exception type and, for me, it makes more sense to look at it as "if you do this, you'll get an exception of this type" instead of "you'll get an exception of this type if you do this".

**Comment by sharwell** [comment]
@paulomorgado In the initial release of this analyzer, it turns out that the content of `<para>` elements is ignored, so a workaround (one I would actually encourage per #603) is to write it like this:

```C#
/// <exception cref="System.ArgumentException">
/// <para>If settings are required and <paramref name="settings" /> is <see langword="null" />.</para>
/// <para>-or-</para>
/// <para>If any of the required settings is not in <paramref name="settings" /> or has invalid values.</para>
/// </exception>
```

**Comment by x3ntrix** [comment]
We have encountered false-positives for the author tags in our code after updating to 1.1.0-beta007.

        /// <summary>
        ///     Gets or sets the name.
        /// </summary>
        /// <author> john.doe@example.com </author>
        public string Name { get; set; }

The warning says that the documentation (in this case the mail address) should end with a period.

**Comment by sharwell** [comment]
@x3ntrix Can you open a separate issue for that? I'm currently working on resolving this bug and likely won't get to that in the same pull request.
